src/game/etj_commands.cpp

```diff
@@ -1254,13 +1254,22 @@ bool Map(gentity_t *ent, Arguments argv)
 		return false;
 	}
 
-	if (!MapExists(argv->at(1)))
+	std::string requestedMap = argv->at(1);
+	boost::to_lower(requestedMap);
+
+	if (!MapExists(requestedMap))
+	{
+		ChatPrintTo(ent, "^3map: ^7map " + requestedMap + " is not on the server.");
+		return false;
+	}
+
+	if (strstr(Q_strlwr(g_blockedMaps.string), requestedMap.c_str()) != nullptr)
 	{
-		ChatPrintTo(ent, "^3map: ^7map " + argv->at(1) + " does not exist.");
+		ChatPrintTo(ent, "^3map: ^7map " + requestedMap + " is blocked.");
```

Comment:
`"map: oasis cannot be played on this server."` would be a clearer way to convey this to the user.